### PR TITLE
gangway: Add ListJobExecutions implementation

### PR DESCRIPTION
**Pull request by diconico07**
This PR actually implements the yet unimplemented but defined `ListJobExecutions` method for gangway.

As I didn't find any specific documentation about how it should behave with regards to authentication, I chose to do as for the `GetJobExecution` and not enforce the `allowed_jobs_filters` for a RO operation.

### CODE REVIEW

File: pkg/gangway/gangway.go [hunk 1]
```diff
@@ -173,6 +173,75 @@ func TranslateProwJobType(prowJobType prowcrd.ProwJobType) JobExecutionType {
...
+func (gw *Gangway) ListJobExecutions(ctx context.Context, ljer *ListJobExecutionsRequest) (*JobExecutions, error) {
+	labelMap, err := metav1.LabelSelectorAsMap(getListRequestLabelSelector(ljer))
+	if err != nil {
+		logrus.WithError(err).Debug("could not convert label selector to map")
+	}
+	timeoutSeconds := int64(LIST_TIMEOUT) // increasing the timeout for large clusters
+	options := metav1.ListOptions{
+		LabelSelector:  labels.SelectorFromSet(labelMap).String(),
+		TimeoutSeconds: &timeoutSeconds,
+	}
```
Comment (Prucek) [inline pkg/gangway/gangway.go:191]: Could you factor this part out? It is the same as in the BulkJobStatusChange

File: pkg/gangway/gangway.go [hunk 2]
```diff
@@ -173,6 +173,75 @@ func TranslateProwJobType(prowJobType prowcrd.ProwJobType) JobExecutionType {
...
+	prowJobCRs, err := gw.ProwJobClient.List(context.TODO(), options)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to list ProwJobs")
+	}
+
+	var jobList []*JobExecution
+	wantedState := translateJobExecutionStatus(ljer.Status)
```
Comment (Prucek) [inline pkg/gangway/gangway.go:198]: You don't need this method, you can do it the other way around

File: pkg/gangway/gangway.go [hunk 3]
```diff
@@ -173,6 +173,75 @@ func TranslateProwJobType(prowJobType prowcrd.ProwJobType) JobExecutionType {
...
+	for _, pj := range prowJobCRs.Items {
+		if wantedState != "" && pj.Status.State != wantedState {
```
Comment (Prucek) [inline pkg/gangway/gangway.go:200]:
```suggestion
		if  TranslateProwJobStatus(&pj.Status) != ljer. Status {
```

**Review by Prucek — COMMENTED** [review]
Could you also maybe add an integration test (/test/integration/test/gangway_test.go) to test if the List is really working?

**Comment by diconico07** [comment]
Added integration tests, and applied the modifications you asked for

File: pkg/gangway/gangway.go [hunk 4]
```diff
@@ -261,6 +294,19 @@ func getRequestLabelSelector(request *BulkJobStatusChangeRequest) *metav1.LabelS
 	return labelSelector
 }
 
+func getListOptions(selector *metav1.LabelSelector) metav1.ListOptions {
+	labelMap, err := metav1.LabelSelectorAsMap(selector)
+	if err != nil {
+		logrus.WithError(err).Debug("could not convert label selector to map")
```
[the remark this answers is not in the document]
Comment (Prucek) [reply pkg/gangway/gangway.go:300]: Why not? What is the problem with that?

**Comment by Prucek** [comment]
> Added integration tests, and applied the modifications you asked for

@diconico07 Sorry but I don't see an integration test in this PR, and I think the documentation part (pkg/plugins/plugin-config-documented.yaml) added is a mistake?

**Comment by diconico07** [comment]
@Prucek indeed, `git add` mistake, added the wrong file, now the integration test is here

File: pkg/gangway/gangway.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (Prucek) [reply pkg/gangway/gangway.go:300]: so you want to just end with 
```logrus.WithError(err).Fatal("could not convert label selector to map")``` ? Or do you mean something else?

[the remark this answers is not in the document]
Comment (Prucek) [reply pkg/gangway/gangway.go:300]: @diconico07 could you resolve this then?

[the remark this answers is not in the document]
Comment (diconico07) [reply pkg/gangway/gangway.go:300]: Done, I explicitly set it to an empty map if we get an error.

**Comment by Prucek** [comment]
From my site, it's:
/lgtm
Thanks @diconico07 for working on it. 
@petr-muller could you take a look at it for approval?

**Review by petr-muller — APPROVED** [review]
Thanks @diconico07 for the contribution and @Prucek for the review! :+1:

**Pull request merged**